client.go

```diff
@@ -141,6 +142,22 @@ func (c *client) deleteServicePrincipal(ctx context.Context, spObjectID string,
 	return c.provider.DeleteServicePrincipal(ctx, spObjectID, permanentlyDelete)
 }
 
+// generateAssignmentIDsFromRole pre-generates UUIDs to be
+// provided to assignRoles in case we need to rollback
+func (c *client) generateAssignmentIDsFromRole(role *roleEntry) ([]string, error) {
```

Comment:
we're not actually using the roles, but only the length of the array. Maybe rework this to just be a simple function, `generateUUIDs`, that generates and returns a list of UUIDs of a certain length?

Reply:
Good call! Updated in latest commit